libraries/rush-lib/src/utilities/Utilities.ts

```diff
@@ -290,19 +291,35 @@ export class Utilities {
    * NOTE: The filenames can also be paths for directories, in which case the directory
    * timestamp is compared.
    */
-  public static isFileTimestampCurrent(dateToCompare: Date, inputFilenames: string[]): boolean {
-    for (const inputFilename of inputFilenames) {
-      if (!FileSystem.exists(inputFilename)) {
-        return false;
-      }
+  public static async isFileTimestampCurrentAsync(
+    dateToCompare: Date,
+    inputFilePaths: string[]
+  ): Promise<boolean> {
+    let anyAreOutOfDate: boolean = false;
+    await Async.forEachAsync(
+      inputFilePaths,
+      async (filePath) => {
+        if (!anyAreOutOfDate) {
+          let inputStats: FileSystemStats | undefined;
+          try {
+            inputStats = FileSystem.getStatistics(filePath);
```

Comment:
```suggestion
            inputStats = await FileSystem.getStatisticsAsync(filePath);
```
Isn't that why this function used `Async`?